### PR TITLE
Erase by swap in filter.

### PR DESCRIPTION
**Pull request by Shylock-Hg**
From O(N^2) to O(N) but lost the order of elements (which we don't need in the modified executor in fact).

### CODE REVIEW

**Review by nevermore3 — APPROVED** [review]

**Review by yixinglu — DISMISSED** [review]

File: src/context/test/IteratorTest.cpp [hunk 1]
```diff
@@ -399,6 +399,66 @@ TEST(IteratorTest, GetNeighbor) {
...
         EXPECT_EQ(result.values, expected);
     }
+    // unstableErase
+    {
+        GetNeighborsIter iter(val);
+        size_t i = 0;
+        while (iter.valid()) {
+            ++i;
+            if (i % 2 == 0) {
+                iter.erase();
```
Comment (yixinglu) [inline src/context/test/IteratorTest.cpp:409]: unstableErase() ?

Comment (Shylock-Hg) [reply src/context/test/IteratorTest.cpp]: Good catch. Fixed.

File: src/context/Iterator.h [hunk 2]
```diff
@@ -179,6 +191,11 @@ class DefaultIter final : public Iterator {
         counter_--;
     }
 
+    void unstableErase() override {
+        DLOG(ERROR) << "Unimplemented default iterator.";
```
Comment (CPWstatic) [inline src/context/Iterator.h:195]: Would it be better if we use DCHECK here?

**Review by CPWstatic — APPROVED** [review]
LGTM.

**Comment by CPWstatic** [comment]
Emmm, we need implement the move constructor for logicalrows.

**Pull request merged**